src/resources/lib/pdfjs/web/viewer.js

```diff
@@ -2384,44 +2384,98 @@ const PDFViewerApplication = {
 
       eventBus._on("pagechanging", _boundEvents.reportPageStatsPDFBug);
     }
-  },
+
+      //// /////// /// // // //// / // // / // / // // / / // /
+      //// /////// /// // // //// / // // / // / // // / / // /
+      //// /////// /// // // //// / // // / // / // // / / // /
+      //// /////// /// // // //// / // // / // / // // / / // /
+      //// /////// /// // // //// / // // / // / // // / / // /
+      //// /////// /// // // //// / // // / // / // // / / // /
+
+      const pageRenderedExtract = async (ev) => {
+
+
+          try {
+
+              if (ev.pageNumber === 1) {
+
+                  const page = ev.source;
+                  const img = page?.canvas?.toDataURL("image/png");
+                  const doc = page?.annotationLayerFactory?.pdfDocument;
+                  const metadata = await doc.getMetadata();
+                  const numberofpages = doc?.numPages;
+                  const numberOfPagesChecked = typeof numberofpages === "number" ? numberofpages : 0;
+
+                  const data = {
+                      ...metadata,
+                      img,
+                      numberofpages: numberOfPagesChecked,
+                  };
+
+                  const str = JSON.stringify(data);
+
+                  const ipc = require('electron').ipcRenderer;
+                  ipc.send("pdfjs-extract-data", str);
+
+                  eventBus._off("pagerendered", pageRenderedExtract);
+              }
+          } catch (e) {
+              console.log("ERROR TO EXTRACT COVER AND METADATA FROM PDF");
+              console.log("ERROR", e);
+          }
+
+      }
+
+
+      eventBus._on("pagerendered", pageRenderedExtract);
+
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+        //// /////// /// // // //// / // // / // / // // / / // /
+
+
+    },
```

Comment:
The above change is the only significant code modification. I will port it to EDRLab's fork, and I will build and publish a new version of PDF.js including this new feature. Then, I will update this PR to include the correct / latets PDF.js version.

Reply:
References:
https://github.com/edrlab/pdf.js/blob/b04b7e5bf8ae5efe8a6e0c6d98e042f89ad928d4/web/ui_utils.js#L845-L846

Then we will update the README to reference the correct Git commit hash(es): 
https://github.com/edrlab/thorium-reader/blob/develop/src/resources/lib/pdfjs/README.md

(no need to update the TypeScript type definitions, as the JS source code has not changed: https://github.com/edrlab/thorium-reader/blob/develop/src/typings/pdf.js/README.md )

Reply:
In progress: https://github.com/edrlab/pdf.js/commit/d9d169afca9970661e750b052cdf64213265dbff
(just looking into optimizing the IPC data payload using Structured Clone Algorithm rather than string serialization (image buffer))

Reply:
I am trying this: https://github.com/edrlab/pdf.js/commit/fd2f4ddf4911884ced3cf4fdf6bd1b53dddbcc8b

Reply:
Works great :)
Super fast. Low memory overhead too.